### PR TITLE
Travel

### PR DESCRIPTION
**Pull request by WojtekMs**

### CODE REVIEW

**Review by kuchichan — CHANGES_REQUESTED** [review]
Please change pr to homework/oop1

**Comment by WojtekMs** [comment]
Base branch changed to homework/oop1 from master :)

File: shm/src/Travel.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        enterCoordinates();
+    }
+    auto daysOfTravel = countDaysOfTravel(player->getSpeed());
+    incrementDays(daysOfTravel);
+    changeCurrentPositionToChosenIsland();
+    std::cout << "Time of travel: " << daysOfTravel << "\n";
+}
+
+void Travel::displayPossibleOptions() {
+    for (auto elem : map_->getAllIslands()) {
```
Comment (kuchichan) [inline shm/src/Travel.cpp:27]: Maybe some std::algorithm? find_if + remove (just to excercise ;d )

File: shm/src/Travel.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    std::cin >> coordinate_x >> coordinate_y;
+}
+
+void Travel::displayCurrentPosition() {
+    std::cout << "Your current position: " << map_->getCurrentPosition()->getCoordinates() << "\n";
+}
+
+void Travel::incrementDays(size_t count) {
+    for (size_t i = 0; i < count; ++i) {
+        TimeServiceLocator::getTime()->operator++();
```
Comment (kuchichan) [inline shm/src/Travel.cpp:46]: Hmm this is somewhat suspicious - did you ensure, that during travel days are also changed on different objects ( i mean we are calling next day method on other observers) - we want to travel in space, not in time :)

**Review by kuchichan — COMMENTED** [review]
Cool!

File: shm/src/Travel.cpp (hunk 2 above)
Comment (kuchichan) [reply shm/src/Travel.cpp:46]: I have checked that - its Ok.

**Review by kuchichan — APPROVED** [review]

File: shm/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+notes.txt
```
Comment (kuchichan) [inline shm/.gitignore]: There is a .gitignore one directory up (above shm) - plase append that line there.

**Pull request merged**